node/src/chain/acceptor.rs

```diff
@@ -125,17 +168,24 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
         let mut task = self.task.write().await;
         let (_, public_key) = task.keys.clone();
 
-        // Verify Block Header
-        self.verify_block_header(&public_key, &blk.header).await?;
-
         let mut mrb = self.mrb.write().await;
         let mut provisioners_list = self.provisioners_list.write().await;
 
+        // Verify Block Header
+        verify_block_header(
+            self.db.clone(),
+            &mrb.header.clone(),
+            provisioners_list.clone(),
+            &public_key,
+            &blk.header,
+        )
+        .await?;
+
         // Reset Consensus
         task.abort_with_wait().await;
 
         // Persist block in consistency with the VM state update
-        let updated_provisioners = {
+        {
```

Comment:
Any specific reason to move the update of the provisioners inside this scope?

Reply:
Yes, for consistency with updating the mrb field.